compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantValueResolver.java

```diff
@@ -667,8 +672,8 @@ private void updateConstantType(BConstantSymbol symbol, BLangExpression expr, Sy
             return;
         }
 
-        if (resolvedType.getKind() == TypeKind.INTERSECTION && isListOrMapping(type.tag)) {
-            expr.setBType(((BIntersectionType) resolvedType).effectiveType);
+        if (resolvedType.getKind() == TypeKind.RECORD && isListOrMapping(type.tag)) {
```

Comment:
Was this intersection only for records? Not for lists (tuples)?

Reply:
Since this fix is only for the constant annotation. Then record would be the only possible case not tuple. Right?

Reply:
I was thinking about when there's a list as a member (field) of the annotation? I don't think we get here for that though.

```ballerina
const annotation map<map<string>[]> annot on source var;

@annot {
    x: [{a: "hello", b: "world"}]
}
var val = 1;
```